### PR TITLE
add a trailing whitespace lint

### PR DESCRIPTION
**Pull request by efx**
I was curious about what [util/exercise](https://github.com/exercism/rust/tree/main/util/exercise) and noticed the `Cargo.toml` has trailing whitespace.
So I took the opportunity to add a lint in CI too.
[This](https://github.com/exercism/rust/runs/1973793069#step:8:4) is an example failure.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -74,7 +77,7 @@ jobs:
 
   # stolen from https://raw.githubusercontent.com/exercism/github-actions/main/.github/workflows/shellcheck.yml
   shellcheck:
-    name: shellcheck internal tooling
+    name: shellcheck internal tooling lint
```
Comment (efx) [inline .github/workflows/tests.yml:80]: Renamed this for consistency. I should also rename our `markdown_lint.sh` to `lint_markdown.sh`. It seems most of our internal tooling follows a `category-subcategory` naming scheme.

File: bin/lint_trailing_spaces.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+# Report any .toml files that have trailing white space
+# so user can fix them
+set -eo pipefail
+
+files=$(find . -name '*.toml' -exec grep -l '[[:space:]]$' {} \;)
```
Comment (coriolinus) [inline bin/lint_trailing_spaces.sh:7]:
```suggestion
files=$(find . -iname '*.toml' -o -iname '*.rs' -o -iname '*.md' -exec grep -l '[[:space:]]$' {} \;)
```

`-name` -> `-iname` just makes the match case-insensitive, in case someone ever creates `IMPORTANT.MD` or something.

I think that we want to lint rust and markdown files as well.

File: util/exercise/src/main.rs [hunk 3]
```diff
@@ -102,7 +102,7 @@ fn process_matches(matches: &ArgMatches<'_>) -> Result<()> {
         }
 
         ("", None) => {
-            println!("No subcommand was used.\nUse init_exercise --help to learn about the possible subcommands.");
+            println!("No subcommand was used.\nUse exercise --help to learn about the possible subcommands.");
```
Comment (coriolinus) [inline util/exercise/src/main.rs:105]: This change makes sense, but seems pretty unrelated to the rest of this PR.

**Review by coriolinus — APPROVED** [review]
Good idea, makes sense.

File: bin/lint_trailing_spaces.sh (hunk 2 above)
Comment (efx) [reply bin/lint_trailing_spaces.sh:7]: Awesome! So our current markdown lint will cover `.md` files so perhaps we only do `.rs` and `.sh`?

Comment (coriolinus) [reply bin/lint_trailing_spaces.sh:7]: Sure, works for me.

File: util/exercise/src/main.rs (hunk 3 above)
Comment (efx) [reply util/exercise/src/main.rs:105]: I agree. How do you feel about documenting something like a Unix philosophy for our pull requests?

Keep it small.
One logical change or something.
I can take a stab for #1111.

Comment (coriolinus) [reply util/exercise/src/main.rs]: Yeah, "one logical change" is the best expression of what I want to see in a PR. It's not necessarily small--there have been good, useful PRs which added hundreds of lines of code and took months to resolve (looking at you, #653)--but life is easier when PRs are focused.
